docs/sphinx/hpx-functions.rst

```diff
@@ -0,0 +1,102 @@
+.. _HPX Functions:
+
+=============
+HPX Functions
+=============
+
+This document describes the prototypes, the allowable arguments of the listed
+HPX functions.
+
+async
+=====
+
+Async in C++ allows running asynchronous programming. Async programming is useful
+in compute heavy implementations. Basic async function would look like
+
+.. code-block:: text
+   #include <future>
+   using namespace hpx;
+   int main() {
+    future<int> f = async([]() {});
+    int val = f.get();
+   }
+
+You can chain multiple async operations. For implementing chain async programming
+``objFuture.then(...).then(...).then(...).then(...).then(...)``.
+
+Implementing async functions should have input parameters as
+
+- Lambda function
+- Executor function
+- Launch policy
+
+then
```

Comment:
FWIW, you can find some description of future::then here: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/lcos/future.hpp#L671-L702

Reply:
I added in my words, do you suggest writing in these lines?

Reply:
Much better now. One can pass launch policies or executor instances to `future::then` as well, btw. I linked the doxygen comments for this before: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/lcos/future.hpp#L671-L702.

Reply:
WDYM?! I didn't change ``then`` documentation?!  I asked if you would want ``then`` documentation written as per these lines?